src/main/webapp/src/app/ranking/ranking-item/ranking-item.component.ts

```diff
@@ -23,8 +23,11 @@ export class RankingItemComponent implements OnInit{
     }
 
     // This prevent the tests from failing with "Cannot read property '0' of undefined"
-    if (this.hiddenGem.types != null && this.hiddenGem.types[0] != null) {
-      this.type = this.hiddenGem.types[0];
+    if (this.hiddenGem.types != null) {
+      if (this.hiddenGem.types[0] != null && (this.hiddenGem.types[0] === 'restaurant' || this.hiddenGem.types[0] === 'cafe'))
+        this.type = this.hiddenGem.types[0];
+      else if (this.hiddenGem.types[1] != null && (this.hiddenGem.types[1] === 'restaurant' || this.hiddenGem.types[1] === 'cafe'))
```

Comment:
what if there is a business that has both types 'restaurant' and 'cafe' and it will be both type[1] and type[0] but will pick the first statement because the order of [0] is before [1]? or is it preventing duplicates by ensuring we only pick one business either cafe or restaurant one time?